### ISSUE TITLE
[Layer Issue] Image Resampling - DownSampling2dLayer is KO

### ISSUE THREAD

**Issue by DEKHTIARJonathan** [tensorlayer/tensorlayer#803]
The layer `tl.layers.DownSampling2dLayer` is KO.

```python
import tensorflow as tf
import tensorlayer as tl

tf.logging.set_verbosity(tf.logging.DEBUG)
tl.logging.set_verbosity(tl.logging.DEBUG)

plh = tf.placeholder(tf.float16, [32, 16, 16, 3])

net_in = tl.layers.InputLayer(plh)
print("INPUT SHAPE: %s" % str(net_in.outputs.shape))

net_up = tl.layers.UpSampling2dLayer(net_in, size=(2, 2), is_scale=True, method=0, align_corners=True)
print("UPSAMPLING OUT SHAPE: %s" % str(net_up.outputs.shape))

try:
    net_down = tl.layers.DownSampling2dLayer(net_up, size=(0.5, 0.5), is_scale=True, method=0, align_corners=True)
    print("DOWNSAMPLING OUT SHAPE (float scale): %s" % str(net_down.outputs.shape))

except:
    net_down = tl.layers.DownSampling2dLayer(net_up, size=(2, 2), is_scale=True, method=0, align_corners=True)
    print("DOWNSAMPLING OUT SHAPE (int scale): %s" % str(net_down.outputs.shape))
```

Output:

```python
[TL] InputLayer  input: (32, 16, 16, 3)
INPUT SHAPE: (32, 16, 16, 3)
[TL] UpSampling2dLayer upsample2d_layer: is_scale: True size: (2, 2) method: 0 align_corners: True
UPSAMPLING OUT SHAPE: (32, 32, 32, 3)
[TL] DownSampling2dLayer downsample2d_layer: is_scale: True size: (0.5, 0.5) method: 0, align_corners: True
[TL] DownSampling2dLayer downsample2d_layer: is_scale: True size: (2, 2) method: 0, align_corners: True
DOWNSAMPLING OUT SHAPE (int scale): (32, 64, 64, 3)   # <= ERROR HERE, should be: (None, 16, 16, 3)
```

Impossible to downscale with a float scale factor < 1 or an int scale factor > 1.
Basically `tl.layers.DownSampling2dLayer` and `tl.layers.UpSampling2dLayer` are identical.

**I have fix this issue in PR #755, however this not merged yet.**

Btw, I'm quite surprised that there is no existing unittest for this layer. 
PR #796 updated this layer without fixing the bug...

**Comment by zsdonghao** [comment]
@DEKHTIARJonathan will this bug make other codes fail to run? 
If it is very important, could we make a PR for it?

**Comment by DEKHTIARJonathan** [comment]
Only DownSampling2dLayer is concerned.

It must be a layer not oftenly used. It's broken for a long time 😅

**Comment by zsdonghao** [comment]
fixed by TL 2.0.0
